File: Python/pangram.py
ALFABET = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']
def is_pangram(sentence):
    letters_used = []
    for letter in range(26):
        for char in sentence.casefold():
            # print(ALFABET[letter])
            if char == ALFABET[letter]:
                letters_used.append(ALFABET[letter])
                print(ALFABET[letter])
                print("done")
                break
    if letters_used == ALFABET:
        result = True
    else:
        result = False
    return result

x = is_pangram("the quick brown fox jumps over the lazy dog")
print(x)

File: Ruby/pangram.rb
class Pangram
  
    def self.pangram?(sentence)
      sentence = sentence.downcase
      count = 0
      target = ('abcdefghijklmnopqrstuvwxyz')
      target.each_char do |n|
        sentence.each_char do |m|
          if n == m
            # puts n + m
            count += 1
            break
            # puts count
          end
        end
      end
      puts count
      puts target.length
      return count == (target.length)

    end
  end

    puts Pangram.pangram?('')
    puts Pangram.pangram?('abcdefghijklmnopqrstuvwxyz')
    puts Pangram.pangram?('the quick brown fox jumps over the lazy dog')
    puts Pangram.pangram?('a quick movement of the enemy will jeopardize five gunboats')
    puts Pangram.pangram?('five boxing wizards jump quickly at it')
    puts Pangram.pangram?('the_quick_brown_fox_jumps_over_the_lazy_dog')
    puts Pangram.pangram?('the 1 quick brown fox jumps over the 2 lazy dogs')
    puts Pangram.pangram?('7h3 qu1ck brown fox jumps ov3r 7h3 lazy dog')
    puts Pangram.pangram?('"Five quacking Zephyrs jolt my wax bed."')
    puts Pangram.pangram?('the quick brown fox jumps over with lazy FX')